core/lib/CMakeLists.txt

```diff
@@ -38,6 +38,9 @@ set(SOCI_STATIC ON)
 set(SOCI_SHARED OFF CACHE BOOL "USE ONLY STATIC" FORCE)
 
 add_subdirectory(soci)
+# Fix undefined symbol related to dl library on Unix system
+# The dl library is linked here to avoid to modify vendor cmake files
+target_link_libraries(soci_core_static ${CMAKE_DL_LIBS})
```

Comment:
Shouldn’t that be gated by the platform, at least `if (UNIX)`?

Reply:
You're right !

Reply:
Can confirm (had a hackier version of the PR for a while) :

`  target_link_libraries(${SOCI_CORE_TARGET_STATIC} dl)` fails on Windows
`  target_link_libraries(${SOCI_CORE_TARGET_STATIC} ${CMAKE_DL_LIBS})` works on Windows

Reply:
(Well, CI too can confirm, I suppose)

Reply:
The [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_DL_LIBS.html) doesn’t say anything about what it should do if it’s not on UNIX, which is why I got a bit scared there. I guess it should be fine…